scipy/ndimage/tests/test_filters.py

```diff
@@ -692,67 +693,75 @@ def test_gauss_memory_overlap(self):
         ndimage.gaussian_filter(input, 1.0, output=input)
         assert_array_almost_equal(output1, input)
 
+    @pytest.mark.parametrize(('filter_func', 'size0', 'size'),
+                             [(ndimage.gaussian_filter, 0, 1.0),
+                              (ndimage.uniform_filter, 1, 3.0),
+                              (ndimage.minimum_filter, 1, 3.0),
+                              (ndimage.maximum_filter, 1, 3.0)])
     @pytest.mark.parametrize(
         'axes',
         tuple(itertools.combinations(range(-3, 3), 1))
         + tuple(itertools.combinations(range(-3, 3), 2))
         + ((0, 1, 2),))
-    def test_gauss_axes(self, axes):
+    def test_filter_axes(self, filter_func, size0, size, axes):
```

Comment:
Changes to this one are probably pretty easy to follow. Now we just need to parameterize over valid combinations of function name, the value of the `size`/`sigma` argument for the axis that is _not_ supposed to be filtered, the value of the `size`/`sigma` argument for the axes that _are_ filtered.

Reply:
`size`/`size0` may be a little confusing for Gaussian but I think it is okay. `sigma` is related to the size of the filter even if it isn't directly an integer size...

Reply:
I thought about picking a word for size that is different from both, like `scale`?

Reply:
```suggestion
    def test_filter_axes(self, filter_func, size0, size, axes):
        # Note: `size` is called `sigma` in `gaussian_filter`
```